todo-app/src/App.js

```diff
@@ -1,13 +1,152 @@
-import React, { Component } from 'react';
+import React, { Component } from 'react'
+import todos from './todosTable'
+import Todo from './todo'
+import './App.css'
 
 class App extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      todos: [],
+      newTodo: {
+        id: '',
+        description: '',
+        stage: ''
+      },
+      updatedTodo: {
+        description: '',
+      },
+      visibility: false,
+      clicked: ''
+    }
+
+    this.changeTodoStage = this.changeTodoStage.bind(this)
+    this.captureUserInput = this.captureUserInput.bind(this)
+    this.onFormSubmit = this.onFormSubmit.bind(this)
+    this.deleteTodo = this.deleteTodo.bind(this)
+    this.updateTodo = this.updateTodo.bind(this)
+    this.changeButtonsVisibility = this.changeButtonsVisibility.bind(this)
+  }
+
+  componentDidMount() {
+    setTimeout(() => {
+      this.setState({ todos })
+    }, 1000)
+  }
+
+  changeTodoStage(todoStage, todoID) {
+    const updatedTodos = this.state.todos.map(todo => {
+      if (todoID === todo.id) {
+        return Object.assign({}, todo, { stage: todoStage })
+      }
+      return todo;
+    })
+    this.setState({
+      todos: updatedTodos
+    });
+  }
+
+  // toggle the Visibility of cancel and update buttons
+  changeButtonsVisibility(show, clickedBtn) {
+    this.setState({
+      visibility: show,
+      // save the clicked edit button to render cancel and update buttons exclusively
+      clicked: clickedBtn
+      ///////////////////////////////////////////////////////////////////////////////
+    })
+  }
+//////////////////////////////////////////////////////
```

Comment:
What are these long lines of `////`?

Reply:
to mark the end of change :) ... but as of your response I might have not to use them again :)